src/backend/commands/self/suicide.cpp

```diff
@@ -9,7 +9,10 @@ namespace big
 
 		virtual void execute(const command_arguments&, const std::shared_ptr<command_context> ctx) override
 		{
+			auto godmode_backup = g.self.god_mode;
+			g.self.god_mode     = false;
 			ENTITY::SET_ENTITY_HEALTH(self::ped, 0, 0);
+			g.self.god_mode = godmode_backup;
```

Comment:
This function works just fine without disabling godmode, any reason why this is needed?

Reply:
What are you talking about? I added a hook to SET_ENTITY_HEALTH that sets the 2nd parameter to the player's max health if g.self.god_mode is on, did you even look at the entire commit?